src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdScopeStoreResolver.cs

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Concurrent;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+using OpenIddict.Abstractions;
+using OrchardCore.OpenId.YesSql.Models;
+using OrchardCore.OpenId.YesSql.Stores;
+
+namespace OrchardCore.OpenId.YesSql.Resolvers
+{
+    /// <summary>
+    /// Exposes a method allowing to resolve a scope store.
+    /// </summary>
+    public class OpenIdScopeStoreResolver : IOpenIddictScopeStoreResolver
+    {
+        private readonly ConcurrentDictionary<Type, Type> _cache = new ConcurrentDictionary<Type, Type>();
```

Comment:
Cache of types, can it be static of does it depend on specific tenant configuration?

Reply:
Nope, but I updated the resolvers to use private singleton caches injected via DI (it's the same approach as the one used in OpenIddict itself). See https://github.com/openiddict/openiddict-core/issues/726 for more context.